dali/kernels/CMakeLists.txt

```diff
@@ -24,9 +24,40 @@ collect_headers(DALI_INST_HDRS PARENT_SCOPE)
 collect_sources(DALI_KERNEL_SRCS)
 collect_test_sources(DALI_KERNEL_TEST_SRCS)
 
-cuda_add_library(${dali_kernel_lib} STATIC "${DALI_KERNEL_SRCS}")
+cuda_add_library(${dali_kernel_lib} SHARED ${DALI_KERNEL_SRCS})
+set_target_properties(${dali_kernel_lib} PROPERTIES
+    LIBRARY_OUTPUT_DIRECTORY "${DALI_LIBRARY_OUTPUT_DIR}")
+target_link_libraries(${dali_kernel_lib} PUBLIC ${dali_core_lib})
+target_link_libraries(${dali_kernel_lib} PRIVATE "-Wl,--exclude-libs,${exclude_libs}")
+
+set(exports_map_path "${PROJECT_BINARY_DIR}/stage")
+set(lib_exports "libdali_kernels.map")
+configure_file("${DALI_ROOT}/cmake/${lib_exports}.in" "${exports_map_path}/${lib_exports}")
+add_custom_target(${lib_exports} DEPENDS "${exports_map_path}/${lib_exports}")
+add_dependencies(${dali_kernel_lib} ${lib_exports})
+target_link_libraries(${dali_kernel_lib} PRIVATE -Wl,--version-script=${exports_map_path}/${lib_exports})
 
 if (BUILD_TEST)
-cuda_add_library(${dali_kernel_test_lib} STATIC "${DALI_KERNEL_TEST_SRCS}")
-target_link_libraries(${dali_kernel_test_lib} ${dali_kernel_lib})
-endif()
+  set(test_bin "dali_kernel_test.bin")
+  # TODO(janton): create a test_utils_lib with dali_test_config.cc and other common utilities
+  cuda_add_executable(${test_bin}
+    ${DALI_KERNEL_TEST_SRCS}
+    ${DALI_ROOT}/dali/test/dali_test_config.cc)
+
+  # TODO(janton): Remove dependency with dali_lib
+  target_link_libraries(${test_bin} PUBLIC ${dali_kernel_lib} ${dali_lib})
+  target_link_libraries(${test_bin} PRIVATE gtest dynlink_cuda ${DALI_LIBS})
+  target_link_libraries(${test_bin} PRIVATE "-Wl,--exclude-libs,${exclude_libs}")
+
+  set_target_properties(${test_bin} PROPERTIES
+    RUNTIME_OUTPUT_DIRECTORY ${TEST_BINARY_DIR})
+
+  set(CHECK_GTEST_TARGET "check-kernel-gtest")
+  add_custom_target("${CHECK_GTEST_TARGET}")
+  add_custom_command(
+    TARGET ${CHECK_GTEST_TARGET}
+    WORKING_DIRECTORY ${TEST_BINARY_DIR}
+    COMMAND ${test_bin}
+    DEPENDS ${test_bin})
+  add_dependencies(${CHECK_TARGET} ${CHECK_GTEST_TARGET})
```

Comment:
I think this is also repeated and can be made a function.

Reply:
Maybe it's just me, but I really don't like "CMake dialects" that arise when one puts too much CMake logic in functions - it quickly becomes a mess like OpenCV...

Reply:
On the other hand, seeing and reusing some `CREATE_CHECK_TARGET(test_bin, test_target_name)` can probably simplify, as this is exactly repeating the same stuff all over place. And than you need to adjust it only once if it's needed.

Reply:
Done